examples/sample_runtime/src/main.rs

```diff
@@ -95,12 +96,10 @@ impl SampleRuntime {
 }
 
 impl Runtime for SampleRuntime {
-    fn deploy_artifact(
-        &mut self,
-        artifact: ArtifactId,
-        spec: Vec<u8>,
-    ) -> Box<dyn Future<Item = (), Error = ExecutionError>> {
-        Box::new(self.deploy_artifact(artifact, spec).into_future())
+    fn deploy_artifact(&mut self, artifact: ArtifactId, spec: Vec<u8>) -> oneshot::Receiver {
+        let (tx, rx) = oneshot::channel();
```

Comment:
As I've suggested previously, it can make sense to introduce a `Receiver` constructor with an ready-made value:

```rust
impl Receiver {
    pub fn with_result(result: Result<(), ExecutionError>) -> Self {
        let (tx, this) = channel();
        tx.send(result);
        this
    }
}

// Usage:
Receiver::with_result(self.deploy_artifact(artifact, spec))
```